config-service-change-event-generator/src/main/java/org/hypertrace/config/service/change/event/api/ConfigChangeEventGenerator.java

```diff
@@ -0,0 +1,49 @@
+package org.hypertrace.config.service.change.event.api;
+
+import com.google.protobuf.Value;
+
+/** The interface config change event generator. */
+public interface ConfigChangeEventGenerator {
+
+  /**
+   * Send create notification for newly added config
+   *
+   * @param tenantId tenant Id
+   * @param resourceName resource name
+   * @param resourceNamespace resource namespace
+   * @param context context
+   * @param config newly created config
+   */
+  void sendCreateNotification(
+      String tenantId, String resourceName, String resourceNamespace, String context, Value config);
```

Comment:
Configs may or may not have a context, we should be sure to account for that so we don't have to pass special strings and/or nulls across this API.

Reply:
Coming back to this after, I think this is picking up the special default context string because it's pulling in the docs so far down the stack - we don't want to be publishing that.

Reply:
if context is default context, then it will not be published. Otherwise it will be published?

Reply:
We will need to pass null for context, if it is default context. Isn't it?

Reply:
Or you mean we don't need to send context at all?

Reply:
`public static final String DEFAULT_CONTEXT = "DEFAULT-CONTEXT";`

So this is a perfectly valid style of config, that we just internally use this constant to denote. It doesn't have an ID, so would generally be tenant-wide. When we publish changes in a config like that, we don't want to expose that internal string - instead, we'd ideally make the context field on `ConfigChangeEventKey` optional.